src/ProductConstructionService/ProductConstructionService.Api/Configuration/PcsConfiguration.cs

```diff
@@ -70,18 +70,25 @@ public static string GetRequiredValue(this IConfiguration configuration, string
         builder.AddGitHubClientFactory();
         builder.AddWorkitemQueues(azureCredential, waitForInitialization: initializeService);
 
-        builder.Services.AddSingleton<IMaestroApi>(s =>
+        builder.Services.AddScoped<IMaestroApi>(s =>
         {
-            var config = s.GetRequiredService<IConfiguration>();
-            var uri = config.GetValue<string>("Maestro:Uri")
+            var uri = builder.Configuration.GetValue<string>("Maestro:Uri")
```

Comment:
are there any benefits in using `Configuration.GetValue<string>("..")` over `Configuration[".."]?

Reply:
In the `bool` case we're avoiding the casting, but for strings it seems shorter